### PR TITLE
[UC17#148#149] Added functionality for users to add their own custom decks

### LINKED ISSUE #148: Contattare RPC addDeck da deckView per aggiungere il mazzo personalizzato con token e nome del mazzo scelto [UC17]
_**Dod**_
- [x]  La pull request ha un titolo e descrizione adeguata
- [x]  La pull request è stata collegata con la issue del task
- [x]  I requisiti funzionali del task collegato sono soddisfatti
- [x]  Il codice si compila senza errori
- [x]  Gli Unit Tests sono stati scritti e passati 
- [x]  Gli Unit Tests esistenti sono stati passati
- [x]  Il Code Review è stato fatto
- [ ]  La documentazione è stata aggiornata
- [x]  La pull request è stata accettata e fatto merge sul branch development.

 **Effort effettivo:**

### LINKED ISSUE #149: Aggiungere il mazzo creato alla pagina dei mazzi [UC17]
_**Dod**_
- [x]  La pull request ha un titolo e descrizione adeguata
- [x]  La pull request è stata collegata con la issue del task
- [x]  I requisiti funzionali del task collegato sono soddisfatti
- [x]  Il codice si compila senza errori
- [x]  Gli Unit Tests sono stati scritti e passati 
- [x]  Gli Unit Tests esistenti sono stati passati
- [x]  Il Code Review è stato fatto
- [ ]  La documentazione è stata aggiornata
- [x]  La pull request è stata accettata e fatto merge sul branch development.

 **Effort effettivo:**

### PR DESCRIPTION
**Pull request by alessioarcara**
This PR implements #148, #149 and adds a new method createCustomDeck in the DecksActivity to make an RPC call and add a custom deck. Once the custom deck is successfully created, it is added to the list of decks on the page.
Tests for the new method were also added.

### CODE REVIEW

File: src/main/java/com/aadm/cardexchange/client/views/DecksViewImpl.java [hunk 1]
```diff
@@ -18,9 +21,12 @@
 
 public class DecksViewImpl extends Composite implements DecksView, ImperativeHandleDeck {
     private static final DecksViewImpl.DecksViewImplUIBinder uiBinder = GWT.create(DecksViewImpl.DecksViewImplUIBinder.class);
+    private static final String DEFAULT_CUSTOM_DECK_TEXT = "Write here name for your custom deck";
```
Comment (Matte0Sacc0) [inline src/main/java/com/aadm/cardexchange/client/views/DecksViewImpl.java:24]: I'd only make a minimal change in the default placeholder, It seems to be a little longer and i'd prefer somehing shorter and clearer... Maybe "Name the deck to create", but it's obviously a minor fixing with no importance relative to the code!!!

File: src/main/java/com/aadm/cardexchange/client/presenters/DecksActivity.java [hunk 2]
```diff
@@ -75,12 +92,32 @@ public void onSuccess(Boolean result) {
...
+                    view.displayAlert(((AuthException) caught).getErrorMessage());
+                } else {
+                    view.displayAlert("Internal server error: " + caught.getMessage());
+                }
+            }
+
+            @Override
+            public void onSuccess(Boolean result) {
+                if (result) {
+                    view.displayAddedCustomDeck(deckName);
```
Comment (Matte0Sacc0) [inline src/main/java/com/aadm/cardexchange/client/presenters/DecksActivity.java:115]: Nice attention to details, no need of reloading is a better UX

File: src/main/java/com/aadm/cardexchange/client/presenters/DecksActivity.java [hunk 3]
```diff
@@ -50,12 +59,20 @@ public void onSuccess(List<PhysicalCardWithName> result) {
...
+    }
+
     @Override
     public void removePhysicalCardFromDeck(String deckName, PhysicalCard pCard, Consumer<Boolean> isRemoved) {
-        if (deckName == null || deckName.isEmpty() || pCard == null) {
+        if (checkDeckNameInvalidity(deckName)) {
             view.displayAlert("Invalid deck name");
             return;
         }
+        if (pCard == null) {
```
Comment (Matte0Sacc0) [inline src/main/java/com/aadm/cardexchange/client/presenters/DecksActivity.java:72]: Great to have the error distintion!!!

**Review by Matte0Sacc0 — APPROVED** [review]
You've made a good job, i've really appreciated the attention to details even in simple and quick adjutments for a better UX such as differenciating the errors given by the deletion of a card and the fact tha no page reloading is needed to update the decks list after adding a new one.

File: src/test/java/com/aadm/cardexchange/client/DecksActivityTest.java [hunk 4]
```diff
@@ -58,4 +66,122 @@ public void testFetchUserDeckForSuccess() {
...
+                Arguments.of(new RuntimeException())
+        );
+    }
+
+    @ParameterizedTest
+    @NullAndEmptySource
+    public void testRemovePhysicalCardFromDeckForInvalidDeckName(String input) {
+        mockDecksView.displayAlert(anyString());
+        ctrl.replay();
+        decksActivity.removePhysicalCardFromDeck(input, new PhysicalCard(Game.MAGIC, 111, Status.Good, "This is a valid description"), (Boolean bool) -> {
```
Comment (dstops) [inline src/test/java/com/aadm/cardexchange/client/DecksActivityTest.java:83]: This is a minor note, but to improve variety and thus test coverage, I suggest you use if it's possible the random status and game generation method, _Status.randomStatus()_ and _Game.randomGame()_

**Review by dstops — APPROVED** [review]

**Pull request merged**